woocommerce/woocommerce-gutenberg-products-block
# Move `FormStep` to the components package

## What

This PR is based on https://github.com/woocommerce/woocommerce-blocks/pull/11214 (update/checkbox-component)

Moves the FormattedMonetaryAmount to the components package and updates imports used across the project to reflect the new location:

- `checkout-shipping-methods-block` 
- `checkout-shipping-method-block` 
- `checkout-shipping-address-block`
- `checkout-billing-address-block` 
- `checkout-pickup-options-block`
- `checkout-payment-block` 
- `checkout-order-note-block` 
- `checkout-contact-information-block` 

## Why

As more developers integrate their extensions with WooCommerce Blocks, the case for reusing components grows. Rather than recreating components developers can access `wc.blocksComponents` and get the ones we need.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Add an item to your cart and visit the Checkout block and ensure the form steps are displayed correctly.
2. Edit the page with the Checkout block and ensure the form steps load in the editor too.
3. Select one of the form steps (e.g. the Contact Information block) and disable the "Show step number" option.
<img width="283" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/240b5b79-cab0-4fa4-b8ac-ca18359a7e09">

4. Ensure the number disappears.
5. Save the page and view it on the front end.
6. Ensure your change was saved and the form step without the number displays correctly.
7. Uncheck the "Use same address for billing" checkbox and ensure the billing address form opens. Ensure the number is correct.


## Screenshots or screencast



N/A

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Make the `FormStep` component available in the `@woocommerce/blocks-components` package.
